src/parser/index.ts

```diff
@@ -214,3 +228,48 @@ function extractTokens(ctx: Context) {
     return /^[^\w$]$/iu.test(c);
   }
 }
+
+/**
+ * Extracts style source from a SvelteStyleElement and parses it into a PostCSS AST.
+ */
+function parseStyleAst(
+  styleElement: SvelteStyleElement,
+  ctx: Context
+): Root | null {
+  if (!styleElement.endTag) {
+    return null;
+  }
+  let lang = "css";
+  for (const attribute of styleElement.startTag.attributes) {
+    if (
+      attribute.type === "SvelteAttribute" &&
+      attribute.key.name === "lang" &&
+      attribute.value.length > 0 &&
+      attribute.value[0].type === "SvelteLiteral"
+    ) {
+      lang = attribute.value[0].value;
+    }
+  }
+  let parseFn: Parser<Root> | undefined = postcss.parse;
+  switch (lang) {
+    case "css":
+      parseFn = postcss.parse;
+      break;
+    case "scss":
+      parseFn = SCSSparse;
+      break;
+    default:
+      console.warn(`Unknown <style> block language "${lang}".`);
+      parseFn = undefined;
+  }
+  if (parseFn !== undefined) {
+    const styleCode = ctx.code.slice(
+      styleElement.startTag.range[1],
+      styleElement.endTag.range[0]
+    );
+    return parseFn(styleCode, {
```

Comment:
Does the node's location return the location of the svelte file?
I think it makes more sense to return the correct location in advance rather than having the rule calculate the offset location.

Reply:
Changing the offset later would be a breaking change, so could you include it in this PR?

Reply:
I think that it is good to do what is calculated by `convertPostCSSNodeToESLintNode`. But, we can assign directly to `node.source`.